ckan/views/user.py

```diff
@@ -748,10 +748,18 @@ def _prepare(self, id: str) -> tuple[Context, dict[str, Any]]:
         except logic.NotAuthorized:
             base.abort(403, _(u'Unauthorized to reset password.'))
 
+        # Ignore auth for the user_show action only.
+        # user_show will fail for anon users if
+        # `ckan.auth.public_user_details` is False
+        context[u'ignore_auth'] = True
         try:
             user_dict = logic.get_action(u'user_show')(context, {u'id': id})
         except logic.NotFound:
             base.abort(404, _(u'User not found'))
+
+        # remove ignore auth for any future use of this context.
+        del context[u'ignore_auth']
```

Comment:
I tend to create a new context e.g.
```python
show_context = dict(context, ignore_auth=True)
```
instead of inserting and removing keys. We have issues where actions and auth functions modify the context that gets returned and creating a new context dict makes sure that doesn't lead to difficult to trace bugs

Reply:
Not sure I understand this, because we're explicitly getting the user_obj from the context for further use, and then we're returning that context to the caller. So we're clearly modifying the context in `user_show`, and those side effects might be used downstream. 

```
        context[u'ignore_auth'] = True
        try:
            user_dict = logic.get_action(u'user_show')(context, {u'id': id})
        except logic.NotFound:
            base.abort(404, _(u'User not found'))

        # remove ignore auth for any future use of this context.
        del context[u'ignore_auth']

        user_obj = context[u'user_obj']
        g.reset_key = request.args.get(u'key')
        if not mailer.verify_reset_link(user_obj, g.reset_key):
            msg = _(u'Invalid reset key. Please try again.')
            h.flash_error(msg)
            base.abort(403, msg)
        return context, user_dict
```